### PR TITLE
add drop_down menu for selecting module

### PR DESCRIPTION
**Pull request by CorneliaKelinske**

### CODE REVIEW

File: lib/ex_quiz_it_web/live/quiz_live.html.heex [hunk 1]
```diff
@@ -3,6 +3,15 @@
     This is EX_QUIZ_IT!
   </h1>
 
+  <form phx-change="select">
+    <select name="module">
+      <option value={@module}><%= @module %></option>
```
Comment (alanvardy) [inline lib/ex_quiz_it_web/live/quiz_live.html.heex:8]: Nice

File: lib/ex_quiz_it/questions/enums.ex [hunk 2]
```diff
@@ -1,7 +1,7 @@
 defmodule ExQuizIt.Questions.Enums do
   @moduledoc false
-  def q_and_a do
-    {:docs_v1, _, :elixir, _, _, _, functions} = Code.fetch_docs(Enum)
+  def q_and_a(module \\ Enum) do
```
Comment (alanvardy) [inline lib/ex_quiz_it/questions/enums.ex]: I wonder if this is a bad idea putting a default argument here. The default is the concern of the LiveView code, not of this module.

File: lib/ex_quiz_it/questions/enums.ex [hunk 3]
```diff
@@ -1,7 +1,7 @@
 defmodule ExQuizIt.Questions.Enums do
```
Comment (alanvardy) [inline lib/ex_quiz_it/questions/enums.ex:1]: This name may be out of date now, it is no longer just for enums right?

File: lib/ex_quiz_it_web/live/quiz_live.ex [hunk 4]
```diff
@@ -5,18 +5,23 @@ defmodule ExQuizItWeb.QuizLive do
   use Phoenix.LiveView, layout: {ExQuizItWeb.LayoutView, "live.html"}
   alias ExQuizIt.Questions.Enums
 
-  @questions Enums.q_and_a()
+  @modules [Enum, String, List]
```
Comment (alanvardy) [inline lib/ex_quiz_it_web/live/quiz_live.ex:8]: Nice, like this pattern

File: lib/ex_quiz_it_web/live/quiz_live.ex [hunk 5]
```diff
@@ -25,15 +30,31 @@ defmodule ExQuizItWeb.QuizLive do
     {:noreply, assign(socket, display_answer: true)}
   end
 
-  def handle_event("next", _value, socket) do
-    q_and_a = Enum.random(@questions)
+  def handle_event("next", _value, %{assigns: %{questions: questions}} = socket) do
+    q_and_a = Enum.random(questions)
```
Comment (alanvardy) [inline lib/ex_quiz_it_web/live/quiz_live.ex:34]: I think that you might be skipping the List.delete step here... is there a double entry in your select list at this point?

**Review by alanvardy — APPROVED** [review]

File: lib/ex_quiz_it/questions/enums.ex (hunk 2 above)
Comment (CorneliaKelinske) [reply lib/ex_quiz_it/questions/enums.ex]: Good catch! I will put the `Enum` in the mount function instead

File: lib/ex_quiz_it/questions/enums.ex (hunk 3 above)
Comment (CorneliaKelinske) [reply lib/ex_quiz_it/questions/enums.ex:1]: Right! Will change in one of the next PRs

File: lib/ex_quiz_it_web/live/quiz_live.ex (hunk 5 above)
Comment (CorneliaKelinske) [reply lib/ex_quiz_it_web/live/quiz_live.ex:34]: ooopsies !

**Pull request merged**